test/src/e2e.dynval/dv.changeParams.test.ts

```diff
@@ -0,0 +1,224 @@
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import * as chai from "chai";
+import { expect } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import { SDK } from "codechain-sdk";
+import * as stake from "codechain-stakeholder-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes, changeParams, defaultParams } from "./setup";
+import { faucetSecret, faucetAddress } from "../helper/constants";
+
+async function getPossibleAuthors(
+    sdk: SDK,
+    blockNumber: number
+): Promise<null | string[]> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getPossibleAuthors", [blockNumber])
+            .then(result => {
+                if (
+                    result === null ||
+                    result.every((str: any) => typeof str === "string")
+                ) {
+                    resolve(result);
+                }
+                return reject(
+                    Error(
+                        `Expected chain_getPossibleAuthors to return an array of string or null but it returned ${result}`
+                    )
+                );
+            })
+            .catch(reject);
+    });
+}
+
+chai.use(chaiAsPromised);
+
+const [, , ...otherDynValidators] = originalValidators;
+const allDynValidators = [...otherDynValidators];
+
+describe("Change commonParams", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const nodes = withNodes(this, {
+        promiseExpect,
+        validators: allDynValidators.map((signer, index) => ({
+            signer,
+            delegation: 5_000,
+            deposit: 10_000_000 - index // tie-breaker
+        })),
+        overrideParams: {
+            termSeconds: 20,
+            maxNumOfValidators: 8
+        }
+    });
+
+    describe("Change term seconds", async function() {
+        it("Term seconds should be changed", async function() {
+            const margin = 1.2;
+            const checkingNode = nodes[0];
+            const termSeconds = 10;
+            const overrideParams = {
+                termSeconds
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
+                ...defaultParams,
+                ...overrideParams
```

Comment:
You can write like this to reduce indirection.
```ts
const changeTxHash = await changeParams(checkingNode, 1, {
    ...defaultParams,
    termSeconds: 10,
```
And with this code, you could have overwritten some parameters. You should do instead:
```ts
const initialParams = {
    ...defualtParams,
    termSeconds: 20,
    maxNumOfValidators: 8
};
const node = withNodes(..., { overrideParams: initialParams });

it(..., async function() {
    const changeTxHash = await changeParams(checkingNode, 1, {
        ...initialParams,
        termSeconds: 10
    });
});
```

Reply:
I applied it.